app/components/QrCodeScanner/QrCodeScanner.jsx

```diff
@@ -29,15 +40,22 @@ export default class QrCodeScanner extends Component<Props> {
       callback(content, this.stopScanner.bind(this))
     })
 
-    Instascan.Camera.getCameras()
+    // since the browser halts while getting usermedia
+    // let the loading animation finish one round and only then continue
+    this.setState({ loading: true })
+    new Promise(resolve => setTimeout(resolve, 900))
+      .then(() => Instascan.Camera.getCameras())
       .then((cameras: Array<Object>) => {
```

Comment:
Do we want to run validation through something like joi on the returned objects perhaps to make sure the cameras in the array are in the right format? Maybe this is already happening in the getCameras() call. If not, might be a good idea

Reply:
Also, as far as using instascan and other libraries, in general the less libraries we use, the better. we will be implementing more security around npm package management moving forward. However, if it is helpful, of course we can use packages

Reply:
> Also, as far as using instascan and other libraries, in general the less libraries we use, the better. we will be implementing more security around npm package management moving forward. However, if it is helpful, of course we can use packages

Indeed I try to use as less packages as possible. This one ain't new though, it's been used up till now. It's not great in terms of maintenance, error handling nor static typing. I think though this type of consideration would be out of scope for this issue, don't you think?

Reply:
Yes, was just bringing it up :)

Reply:
> Do we want to run validation through something like joi on the returned objects perhaps to make sure the cameras in the array are in the right format? Maybe this is already happening in the getCameras() call. If not, might be a good idea

Not sure it's a must in this case since it's an `Instascan` method feeding into another. I changed the original code to Promise chaining in order for the `catch` to attain to any errors in this process.